iff --git a/CMake/OtherTests.cmake b/CMake/OtherTests.cmake
index b3031f7..a5c7b71 100644
--- a/CMake/OtherTests.cmake
+++ b/CMake/OtherTests.cmake
@@ -85,8 +85,6 @@ endif()
 
 unset(CMAKE_TRY_COMPILE_TARGET_TYPE)
 
-if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
-  if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "iOS")
   # only try this on non-apple platforms
 
   # if not cross-compilation...
@@ -131,6 +129,4 @@ if(NOT DEFINED CMAKE_TOOLCHAIN_FILE)
         }
         return 0;
     }" HAVE_POLL_FINE)
-  endif()
-endif()
 
