diff --git a/CMake/OtherTests.cmake b/CMake/OtherTests.cmake
index ed8d28a21..d3b3d55f7 100644
--- a/CMake/OtherTests.cmake
+++ b/CMake/OtherTests.cmake
@@ -86,9 +86,7 @@ endif()
 unset(CMAKE_TRY_COMPILE_TARGET_TYPE)
 
 if(NOT CMAKE_CROSSCOMPILING)
-  if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND NOT ${CMAKE_SYSTEM_NAME} MATCHES "iOS")
-    # only try this on non-apple platforms
-
+  if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "iOS")
     # if not cross-compilation...
     include(CheckCSourceRuns)
     set(CMAKE_REQUIRED_FLAGS "")
